### PR TITLE
p2p/netutil: unittests for addrutil

### PR DESCRIPTION
**Pull request by tianyeyouyou**
add unit tests for `p2p/addrutil`

### CODE REVIEW

File: p2p/netutil/addrutil_test.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := IPToAddr(tt.ip); got != tt.want {
+				t.Errorf("IPToAddr() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestRandomAddr(t *testing.T) {
```
Comment (holiman) [inline p2p/netutil/addrutil_test.go:127]: I think this is a bit overkill. The `RandomAddr` is ~15 LOC, and essentially just generates random bytes which is translated to `ipv4/ipv6`. It is very trivial, and it is also _only_ invoked from tests. So adding a test for it is very meta. Let's just ensure it works, once, then never bother with it again.  So IMO drop this method

File: p2p/netutil/addrutil_test.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
+package netutil
```
Comment (holiman) [inline p2p/netutil/addrutil_test.go:1]: Please add license info (take from some other file)

**Review by holiman — COMMENTED** [review]
Thanks for this! Looks good to me, aside from a couple of small things.

File: p2p/netutil/addrutil_test.go (hunk 2 above)
Comment (tianyeyouyou) [reply p2p/netutil/addrutil_test.go:17]: Added

File: p2p/netutil/addrutil_test.go (hunk 1 above)
Comment (tianyeyouyou) [reply p2p/netutil/addrutil_test.go]: `TestRandomAddr` has been removed

**Review by holiman — APPROVED** [review]
LGTM

File: p2p/netutil/addrutil_test.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package netutil
+
+import (
+	"math/rand"
```
Comment (holiman) [inline p2p/netutil/addrutil_test.go:20]:
```suggestion
```

**Comment by MariusVanDerWijden** [comment]
Could you rebase it on top of current master? 
There was a failing unit test, which was fixed on master at some point

**Review by MariusVanDerWijden — APPROVED** [review]
SGTM

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**